src/parquet/column/column-writer-test.cc

```diff
@@ -0,0 +1,202 @@
+// under the License.
+
+#include <gtest/gtest.h>
+
+#include "parquet/file/reader-internal.h"
+#include "parquet/file/writer-internal.h"
+#include "parquet/column/reader.h"
+#include "parquet/column/writer.h"
+#include "parquet/util/input.h"
+#include "parquet/util/output.h"
+#include "parquet/types.h"
+
+namespace parquet {
+
+using schema::NodePtr;
+using schema::PrimitiveNode;
+
+namespace test {
+
+class TestPrimitiveWriter : public ::testing::Test {
+ public:
+  void SetUpSchemaRequiredNonRepeated() {
+    node_ = PrimitiveNode::Make("int64", Repetition::REQUIRED, Type::INT64);
+    schema_ = std::make_shared<ColumnDescriptor>(node_, 0, 0);
+  }
+
+  void SetUpSchemaOptionalNonRepeated() {
+    node_ = PrimitiveNode::Make("int64", Repetition::REQUIRED, Type::INT64);
+    schema_ = std::make_shared<ColumnDescriptor>(node_, 1, 0);
+  }
+
+  void SetUpSchemaOptionalRepeated() {
+    node_ = PrimitiveNode::Make("int64", Repetition::REPEATED, Type::INT64);
+    schema_ = std::make_shared<ColumnDescriptor>(node_, 1, 1);
+  }
+
+  void SetUp() {
+    values_out_.resize(100);
+    definition_levels_out_.resize(100);
+    repetition_levels_out_.resize(100);
+
+    SetUpSchemaRequiredNonRepeated();
+  }
+
+  std::unique_ptr<Int64Reader> BuildReader() {
+    auto buffer = sink_->GetBuffer();
+    std::unique_ptr<InMemoryInputStream> source(new InMemoryInputStream(buffer));
+    std::unique_ptr<SerializedPageReader> page_reader(
+        new SerializedPageReader(std::move(source), Compression::UNCOMPRESSED));
+    return std::unique_ptr<Int64Reader>(
+        new Int64Reader(schema_.get(), std::move(page_reader)));
+  }
+
+  std::unique_ptr<Int64Writer> BuildWriter(int64_t output_size = 100) {
+    sink_.reset(new InMemoryOutputStream());
+    std::unique_ptr<SerializedPageWriter> pager(
+        new SerializedPageWriter(sink_.get(), Compression::UNCOMPRESSED, &metadata_));
+    return std::unique_ptr<Int64Writer>(new Int64Writer(schema_.get(), std::move(pager),
+          output_size));
+  }
+
+  void ReadColumn() {
+    auto reader = BuildReader();
+    reader->ReadBatch(values_out_.size(), definition_levels_out_.data(),
+        repetition_levels_out_.data(), values_out_.data(), &values_read_);
+  }
+
+ protected:
+  int64_t values_read_;
+
+  // Output buffers
+  std::vector<int64_t> values_out_;
+  std::vector<int16_t> definition_levels_out_;
+  std::vector<int16_t> repetition_levels_out_;
+
+ private:
+  NodePtr node_;
+  format::ColumnChunk metadata_;
+  std::shared_ptr<ColumnDescriptor> schema_;
+  std::unique_ptr<InMemoryOutputStream> sink_;
+};
+
+TEST_F(TestPrimitiveWriter, RequiredNonRepeated) {
+  std::vector<int64_t> values(100, 128);
+
+  // Test case 1: required and non-repeated, so no definition or repetition levels
+  std::unique_ptr<Int64Writer> writer = BuildWriter();
+  writer->WriteBatch(values.size(), nullptr, nullptr, values.data());
+  writer->Close();
+
+  ReadColumn();
+  ASSERT_EQ(values_read_, 100);
```

Comment:
The expected value goes first in googletest (no big deal but it impacts the failure message)